## Ćw 1

In [1]:
import pandas as pd
import numpy as np

In [2]:
# Series z listy — domyślny indeks (0, 1, 2, ...)
oceny = pd.Series([4.5, 3.0, 5.0, 3.5, 4.0])
print(oceny)

0    4.5
1    3.0
2    5.0
3    3.5
4    4.0
dtype: float64


## Zad 1a

In [7]:
# Klucze: 'Anna', 'Jan', 'Ewa', 'Marek', 'Kasia'
# Wartości: 5500, 7200, 4800, 9100, 6300
data_dict = {
    'Anna': 5500, 
    'Jan': 7200, 
    'Ewa': 4800, 
    'Marek': 9100, 
    'Kasia': 6300
}
pensje = pd.Series(data_dict)
print(pensje)

Anna     5500
Jan      7200
Ewa      4800
Marek    9100
Kasia    6300
dtype: int64


## Zad 1b

In [18]:
# 1. Wyświetl imię osoby zarabiającej najwięcej (użyj idxmax())
najbogatszy = pensje.idxmax()
print(najbogatszy)
# 2. Oblicz średnią pensję
średnia = pensje.mean()
print(średnia)
# 3. Wyświetl imiona osób zarabiających powyżej średniej (filtrowanie boolean)
powyzej = pensje > pensje.mean()
osoby_powyzej = pensje[powyzej]
print(osoby_powyzej.to_string())
# 4. Policz ile osób zarabia poniżej 6000 (użyj len() lub .sum())
ponizej = pensje < 6000
osoby_ponizej = pensje[ponizej]
print(len(osoby_ponizej))

Marek
6580.0
Jan      7200
Marek    9100
2


## Zad 1c

In [31]:
# 1. Oblicz pensje po podwyżce 10% dla wszystkich (pensje * 1.10)
podwyzka = pensje * 1,10
print("Pensje po podwyżce:")
print(podwyzka)
# 2. Oblicz pensje po premii 500 zł dla wszystkich (pensje + 500)
premia = pensje + 500
print("Pensje po premii:")
print(premia)
# 3. Oblicz nową pensję = (pensja * 1.10) + 500
nowa_pensja = (pensje * 1.10) + 500
print("Nowa pensja:")
print(nowa_pensja) 
# 4. Wyświetl ranking nowych pensji od najwyższej (użyj sort_values)
ranking = pensje.sort_values
print(ranking)


Pensje po podwyżce:
(Anna     5500
Jan      7200
Ewa      4800
Marek    9100
Kasia    6300
dtype: int64, 10)
Pensje po premii:
Anna     6000
Jan      7700
Ewa      5300
Marek    9600
Kasia    6800
dtype: int64
Nowa pensja:
Anna      6550.0
Jan       8420.0
Ewa       5780.0
Marek    10510.0
Kasia     7430.0
dtype: float64
<bound method Series.sort_values of Anna     5500
Jan      7200
Ewa      4800
Marek    9100
Kasia    6300
dtype: int64>


## Ćw 2

In [43]:
# Dane o produktach sklepu internetowego
sklep = pd.DataFrame({
'produkt': ['Laptop', 'Tablet', 'Smartfon', 'Słuchawki', 'Monitor'],
'cena': [3500, 1800, 2500, 350, 2200],
'stan_magazynowy': [45, 120, 200, 500, 75],
'kategoria': ['Komputery', 'Mobilne', 'Mobilne', 'Akcesoria', 'Komputery']
})

In [44]:
print(sklep.shape)
print(sklep.columns)
print(sklep.dtypes)
print(sklep.head)

(5, 4)
Index(['produkt', 'cena', 'stan_magazynowy', 'kategoria'], dtype='str')
produkt              str
cena               int64
stan_magazynowy    int64
kategoria            str
dtype: object
<bound method NDFrame.head of      produkt  cena  stan_magazynowy  kategoria
0     Laptop  3500               45  Komputery
1     Tablet  1800              120    Mobilne
2   Smartfon  2500              200    Mobilne
3  Słuchawki   350              500  Akcesoria
4    Monitor  2200               75  Komputery>


In [50]:
sklep = pd.DataFrame({
'produkt': ['Laptop', 'Tablet', 'Smartfon', 'Słuchawki', 'Monitor'],
'cena': [3500, 1800, 2500, 350, 2200],
'stan_magazynowy': [45, 120, 200, 500, 75],
'kategoria': ['Komputery', 'Mobilne', 'Mobilne', 'Akcesoria', 'Komputery']
})
# 1. Dodaj kolumnę 'wartosc_magazynu' = cena * stan_magazynowy
sklep['wartosc_magazynu'] = sklep['cena'] * sklep['stan_magazynowy']
# 2. Dodaj kolumnę 'cena_z_vat' = cena * 1.23
sklep['cena_z_vat'] = sklep['cena'] * 1.23
# 3. Wyświetl cały DataFrame z nowymi kolumnami
print(sklep)

     produkt  cena  stan_magazynowy  kategoria  wartosc_magazynu  cena_z_vat
0     Laptop  3500               45  Komputery            157500      4305.0
1     Tablet  1800              120    Mobilne            216000      2214.0
2   Smartfon  2500              200    Mobilne            500000      3075.0
3  Słuchawki   350              500  Akcesoria            175000       430.5
4    Monitor  2200               75  Komputery            165000      2706.0


In [51]:
# Wczytaj dataset tips
url = 'https://raw.githubusercontent.com/mwaskom/seaborn-data/master/tips.csv'
tips = pd.read_csv(url)

## Ćw 2c

In [55]:
print(tips.shape)
print(tips.tail)
print(tips.sample)
print(tips.dtypes)

(244, 7)
<bound method NDFrame.tail of      total_bill   tip     sex smoker   day    time  size
0         16.99  1.01  Female     No   Sun  Dinner     2
1         10.34  1.66    Male     No   Sun  Dinner     3
2         21.01  3.50    Male     No   Sun  Dinner     3
3         23.68  3.31    Male     No   Sun  Dinner     2
4         24.59  3.61  Female     No   Sun  Dinner     4
..          ...   ...     ...    ...   ...     ...   ...
239       29.03  5.92    Male     No   Sat  Dinner     3
240       27.18  2.00  Female    Yes   Sat  Dinner     2
241       22.67  2.00    Male    Yes   Sat  Dinner     2
242       17.82  1.75    Male     No   Sat  Dinner     2
243       18.78  3.00  Female     No  Thur  Dinner     2

[244 rows x 7 columns]>
<bound method NDFrame.sample of      total_bill   tip     sex smoker   day    time  size
0         16.99  1.01  Female     No   Sun  Dinner     2
1         10.34  1.66    Male     No   Sun  Dinner     3
2         21.01  3.50    Male     No   Sun  Dinne

## Ćw 3

In [56]:
url = 'https://raw.githubusercontent.com/mwaskom/seaborn-data/master/penguins.csv'
penguins = pd.read_csv(url)

## Ćw 3a

In [59]:
print(penguins.head(10))
print(penguins.shape)
print(penguins.info())

  species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0  Adelie  Torgersen            39.1           18.7              181.0   
1  Adelie  Torgersen            39.5           17.4              186.0   
2  Adelie  Torgersen            40.3           18.0              195.0   
3  Adelie  Torgersen             NaN            NaN                NaN   
4  Adelie  Torgersen            36.7           19.3              193.0   
5  Adelie  Torgersen            39.3           20.6              190.0   
6  Adelie  Torgersen            38.9           17.8              181.0   
7  Adelie  Torgersen            39.2           19.6              195.0   
8  Adelie  Torgersen            34.1           18.1              193.0   
9  Adelie  Torgersen            42.0           20.2              190.0   

   body_mass_g     sex  
0       3750.0    MALE  
1       3800.0  FEMALE  
2       3250.0  FEMALE  
3          NaN     NaN  
4       3450.0  FEMALE  
5       3650.0    MALE  
6       36

## Zad 3b

In [62]:
print(penguins.describe())
print(penguins.describe(include='all'))
print(penguins.body_mass_g)

       bill_length_mm  bill_depth_mm  flipper_length_mm  body_mass_g
count      342.000000     342.000000         342.000000   342.000000
mean        43.921930      17.151170         200.915205  4201.754386
std          5.459584       1.974793          14.061714   801.954536
min         32.100000      13.100000         172.000000  2700.000000
25%         39.225000      15.600000         190.000000  3550.000000
50%         44.450000      17.300000         197.000000  4050.000000
75%         48.500000      18.700000         213.000000  4750.000000
max         59.600000      21.500000         231.000000  6300.000000
       species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
count      344     344      342.000000     342.000000         342.000000   
unique       3       3             NaN            NaN                NaN   
top     Adelie  Biscoe             NaN            NaN                NaN   
freq       152     168             NaN            NaN                NaN   

## Zad 3c

In [73]:
print(penguins['species'].value_counts())
print(penguins['island'].value_counts())
print(penguins['sex'].value_counts())
print('Rozkład płci jest zbalansowany.')
print(penguins['species'].value_counts(normalize= True))

species
Adelie       152
Gentoo       124
Chinstrap     68
Name: count, dtype: int64
island
Biscoe       168
Dream        124
Torgersen     52
Name: count, dtype: int64
sex
MALE      168
FEMALE    165
Name: count, dtype: int64
Rozkład płci jest zbalansowany.
species
Adelie       0.441860
Gentoo       0.360465
Chinstrap    0.197674
Name: proportion, dtype: float64


## Zad 3d

In [ ]:
print(penguins.isna().sum())
brak = penguins['sex'].isna().sum()
procent = brak/len(penguins) * 100
print(procent)

species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
dtype: int64
3.1976744186046515


## Zad 3e

'Dataset zawiera 344  pingwinów z 3 gatunków."
"Najczęstszy gatunek to Adelie (152 osobników, 44%)"
"Brakuje danych w kolumnach species i island(łącznie 2 braków)"
"Średnia masa ciała to 4201 g (mediana: 4050 g)"
"Pingwiny mieszkają na 3 wyspach, najliczniej na Biscoe"

In [78]:
tips = pd.read_csv('https://raw.githubusercontent.com/mwaskom/seaborn-data/master/tips.csv')
# Dodaj kolumnę z procentem napiwku
tips['tip_pct'] = (tips['tip'] / tips['total_bill'] * 100).round(1)

## Zad 4a

In [ ]:
# 1. Oblicz średni rachunek (total_bill)
sredni_rachunek = tips['total_bill'].mean()
print(f"1. Średni rachunek: {sredni_rachunek:.2f}")

# 2. Oblicz średni napiwek w procentach (tip_pct)
sredni_tip_pct = tips['tip_pct'].mean()
print(f"2. Średni napiwek (%): {sredni_tip_pct:.2f}%")

# 3. Wyświetl który dzień ma najwięcej rachunków
dni_counts = tips['day'].value_counts()
print("\n3. Liczba rachunków w poszczególne dni:")
print(dni_counts)
print(f"Najwięcej rachunków było w: {dni_counts.idxmax()}")

# 4. Policz ile rachunków jest od palaczy
palacze_counts = tips['smoker'].value_counts()
print(f"\n4. Liczba rachunków od palaczy (Yes): {palacze_counts['Yes']}")

# 5. Wyświetl describe() tylko dla kolumny 'tip' — jaka jest mediana napiwku?
print("\n5. Statystyki dla kolumny 'tip':")
print(tips['tip'].describe())
# Mediana to 50% w wynikach describe()
mediana_tip = tips['tip'].median()
print(f"Mediana napiwku wynosi: {mediana_tip}")

# 6. Jaki procent rachunków pochodzi z obiadu (Dinner)?
procent_czasu = tips['time'].value_counts(normalize=True) * 100
print("\n6. Procentowy podział czasu (Lunch vs Dinner):")
print(procent_czasu)
print(f"Procent rachunków z Dinner: {procent_czasu['Dinner']:.1f}%")

# 7. Ile unikalnych rozmiarów grup jest w danych?
unikalne_grupy = tips['size'].nunique()
print(f"\n7. Liczba unikalnych rozmiarów grup: {unikalne_grupy}")
print(f"Konkretne rozmiary grup: {tips['size'].unique()}")

1. Średni rachunek: 19.78594262295082
2. Średni napiwek (%): 16.08%

3. Liczba rachunków w poszczególne dni:
day
Sat     87
Sun     76
Thur    62
Fri     19
Name: count, dtype: int64
Najwięcej rachunków było w: Sat

4. Liczba rachunków od palaczy (Yes): 93

5. Statystyki dla kolumny 'tip':
count    244.000000
mean       2.998279
std        1.383638
min        1.000000
25%        2.000000
50%        2.900000
75%        3.562500
max       10.000000
Name: tip, dtype: float64
Mediana napiwku wynosi: 2.9

6. Procentowy podział czasu (Lunch vs Dinner):
time
Dinner    72.131148
Lunch     27.868852
Name: proportion, dtype: float64
Procent rachunków z Dinner: 72.1%

7. Liczba unikalnych rozmiarów grup: 6
Konkretne rozmiary grup: [2 3 4 1 6 5]
